lib/rules/v-if-else-key.js

```diff
@@ -0,0 +1,299 @@
+ */
+const conditionalFamilies = new Map()
+
+// =============================================================================
+// Rule Helpers
+// =============================================================================
+
+/**
+ * Checks for the presence of a 'key' attribute in the given node. If the 'key' attribute is missing
+ * and the node is part of a conditional family a report is generated.
+ * The fix proposed adds a unique key based on the component's name and count,
+ * following the format '${kebabCase(componentName)}-${componentCount}', e.g., 'some-component-2'.
+ *
+ * @param {VElement} node - The Vue component node to check for a 'key' attribute.
+ * @param {RuleContext} context - The rule's context object, used for reporting.
+ * @param {string} componentName - Name of the component.
+ * @param {string} uniqueKey - A unique key for the repeated component, used for the fix.
+ */
+const checkForKey = (node, context, componentName, uniqueKey) => {
+  if (node.parent && node.parent.type === 'VElement') {
+    const conditionalFamily = conditionalFamilies.get(node.parent)
+
+    if (
+      conditionalFamily &&
+      (utils.hasDirective(node, 'bind', 'key') ||
+        utils.hasAttribute(node, 'key') ||
+        !hasConditionalDirective(node) ||
+        !isConditionalFamilyComplete(conditionalFamily))
+    ) {
+      return
+    }
+
+    context.report({
+      node: node.startTag,
+      loc: node.startTag.loc,
+      messageId: 'requireKey',
+      data: {
+        componentName
+      },
+      fix(fixer) {
+        const afterComponentNamePosition =
+          node.startTag.range[0] + componentName.length + 1
+        return fixer.insertTextBeforeRange(
+          [afterComponentNamePosition, afterComponentNamePosition],
+          ` key='${uniqueKey}'`
+        )
+      }
+    })
+  }
+}
+
+/**
+ * Checks for the presence of conditional directives in the given node.
+ *
+ * @param {VElement} node - The node to check for conditional directives.
+ * @returns {boolean} Returns true if a conditional directive is found in the node or its parents,
+ *   false otherwise.
+ */
+const hasConditionalDirective = (node) =>
+  utils.hasDirective(node, 'if') ||
+  utils.hasDirective(node, 'else-if') ||
+  utils.hasDirective(node, 'else')
+
+/**
+ * Checks whether a conditional family is complete.
+ * A conditional family is considered complete if:
+ * 1. An 'if' condition is present.
+ * 2. All elements of the family (if, else, else-if) share the same rawName.
+ * 3. There exists either an 'else' condition or at least one 'else-if' condition.
+ *
+ * @param {{ if: VElement, elseIf: VElement[], else: VElement | null }} family - The conditional family object with 'if', 'else-if', and 'else' properties.
+ * @returns {boolean} True if the conditional family is complete based on the above criteria, false otherwise.
+ */
+const isConditionalFamilyComplete = (family) => {
+  if (!family || !family.if) {
```

Comment:
`if` cannot be falsy anymore, right?

```suggestion
  if (!family) {
```

Reply:
[a8709d7](https://github.com/vuejs/eslint-plugin-vue/pull/2280/commits/a8709d716d6c9bfd3af247ea594d38e8f68b0183)